<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TimeScanMX</class>
 <widget class="QMainWindow" name="TimeScanMX">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1446</width>
    <height>975</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TimeScanMX</string>
  </property>
  <property name="windowIcon">
   <iconset resource="qchartmx.qrc">
    <normaloff>:/new/prefix1/Graph1-small.png</normaloff>:/new/prefix1/Graph1-small.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="control" native="true">
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="widget_3" native="true">
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Continuous scan</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QCheckBox" name="cont">
             <property name="text">
              <string/>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Scan time</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Update Interval</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="period">
             <property name="suffix">
              <string> sec</string>
             </property>
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="minimum">
              <double>0.002000000000000</double>
             </property>
             <property name="maximum">
              <double>2147483647.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="interval">
             <property name="suffix">
              <string> sec</string>
             </property>
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="minimum">
              <double>0.001000000000000</double>
             </property>
             <property name="maximum">
              <double>1073741823.500000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="signalsList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <widget class="QWidget" name="signalsC">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>362</width>
             <height>748</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="spacing">
             <number>1</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QWidget" name="widget_8" native="true">
              <layout class="QGridLayout" name="signalsL">
               <property name="margin">
                <number>0</number>
               </property>
               <property name="spacing">
                <number>1</number>
               </property>
               <item row="0" column="0">
                <widget class="QLabel" name="label_16">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLabel" name="label_17">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                   <horstretch>1</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>PV or command</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_18">
                 <property name="text">
                  <string>Value</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addSignal">
          <property name="toolTip">
           <string>Add signal to be scanned</string>
          </property>
          <property name="text">
           <string>Add signal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_4" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="spacing">
            <number>1</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="saveName">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Name of the file with the scanned data.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="autoName">
             <property name="toolTip">
              <string>Automatically construct the filename for the data to be saved.</string>
             </property>
             <property name="text">
              <string>Auto</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_5" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <property name="spacing">
            <number>1</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="browseSaveDir">
             <property name="toolTip">
              <string>Browse for the directory to save scan data into.</string>
             </property>
             <property name="text">
              <string>to</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="saveDir">
             <property name="toolTip">
              <string>Directory where the scan data are saved.</string>
             </property>
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="visualize" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitter_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="handleWidth">
           <number>9</number>
          </property>
          <property name="childrenCollapsible">
           <bool>false</bool>
          </property>
          <widget class="QTableWidget" name="dataTable">
           <property name="columnCount">
            <number>1</number>
           </property>
           <column>
            <property name="text">
             <string>Time</string>
            </property>
           </column>
          </widget>
          <widget class="QwtPlot" name="plot">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_7" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Min:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="autoMin">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Auto</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="min">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>9</number>
             </property>
             <property name="minimum">
              <double>-2147483647.000000000000000</double>
             </property>
             <property name="maximum">
              <double>2147483647.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Max:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="autoMax">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Auto</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="max">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="decimals">
              <number>9</number>
             </property>
             <property name="minimum">
              <double>-2147483647.000000000000000</double>
             </property>
             <property name="maximum">
              <double>2147483647.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line_4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="logY">
             <property name="text">
              <string>Logarithmic Y</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="norma">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Normalize</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="showGrid">
             <property name="text">
              <string>Show Grid</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_6" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>1</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
           <item>
            <widget class="QPushButton" name="printResult">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Print the plot.</string>
             </property>
             <property name="text">
              <string>Print</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="saveResult">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>Save the data to file</string>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="qtiResults">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Open with QtiPlot</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="startStop">
          <property name="toolTip">
           <string>Start / Stop the scan.</string>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>period</tabstop>
  <tabstop>cont</tabstop>
  <tabstop>addSignal</tabstop>
  <tabstop>saveName</tabstop>
  <tabstop>autoName</tabstop>
  <tabstop>browseSaveDir</tabstop>
  <tabstop>saveDir</tabstop>
  <tabstop>startStop</tabstop>
  <tabstop>printResult</tabstop>
  <tabstop>saveResult</tabstop>
  <tabstop>qtiResults</tabstop>
  <tabstop>dataTable</tabstop>
  <tabstop>signalsList</tabstop>
 </tabstops>
 <resources>
  <include location="qchartmx.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>autoMin</sender>
   <signal>toggled(bool)</signal>
   <receiver>min</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>900</y>
    </hint>
    <hint type="destinationlabel">
     <x>707</x>
     <y>904</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoMax</sender>
   <signal>toggled(bool)</signal>
   <receiver>max</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>830</x>
     <y>900</y>
    </hint>
    <hint type="destinationlabel">
     <x>1053</x>
     <y>904</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoName</sender>
   <signal>clicked(bool)</signal>
   <receiver>saveName</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>313</x>
     <y>924</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>928</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
